### ISSUE TITLE
google.com 

### ISSUE THREAD

**Issue by Morku** [AdguardTeam/AdguardFilters#76433]
### Issue URL (Ads)
 
https://www.google.com/search?client=firefox-b-d&q=computer+leipzig

 ### Comment
 
 Username: @Morku
 
 > ## There is an Ad result (Anzeige) in the geolocation result. Be aware: this is very rare. It's like a 5-10% chance that you will get it. Just watch my video (in case you are bored) that it took me 12 tries to reproduce it, but I can always reproduce over time. When ever I close the Browser, Cache and Cookie are cleaned. For the dozenth time of searching "computer leipzig", I got the Ad result. The search was done by the Address Bar of Firefox.
 > Die folgenden Informationen werden automatisch generiert. Bitte löschen Sie sie nicht.
 > Custom userscripts:
 > https://userscripts.adtidy.org/release/adguard-extra/1.0/adguard-extra.user.js
 
 ### Screenshots
<details><summary>Screenshot-1:</summary>

![google](https://user-images.githubusercontent.com/6276915/109850719-1a482e80-7c53-11eb-8674-19b7a85ae140.png)

</details><br/>

 ### Videos
<details><summary>Video-1:</summary>

https://user-images.githubusercontent.com/6276915/109851228-bbcf8000-7c53-11eb-8fcd-ad63e9a27c39.mp4

</details><br/>

 ### System configuration
 Information 	value
 AdGuard product: 	AdGuard for Windows v7.5.3
 Browser: 	Firefox
 AdGuard driver: 	WFP
 Stealth mode options: 	Hide your search queries,
 Send Do-Not-Track header,
 Remove X-Client-Data header from HTTP requests,
 Strip URLs from tracking parameters

 Filters: 	Ad Blocking:
 AdGuard Mobile Ads,
 AdGuard Base
 
 Privacy:
 AdGuard Tracking Protection
 
 Social Widgets:
 AdGuard Social Media
 
 Annoyances:
 AdGuard Annoyances
 
 Language-specific:
 AdGuard Russian,
 AdGuard German

**Comment by Alex-302** [comment]
Not reproduced even witohut AdGuard
![image](https://user-images.githubusercontent.com/8361299/110340103-8ea71700-8031-11eb-9c56-63ecc413d95d.png)

@BlazDT please take a look.

**Comment by wout-codes** [comment]
@Morku Not reproduced. But can you test this rule like what you have done in the video? `google.com#?#jsname[role*="link"]:contains(Anzeige)`

**Comment by Morku** [comment]
I still can reproduce. With and without AdGuard. But it always takes about 5-10 tries to get it (also without AdGuard)

<details><summary>Without AG:</summary>

![Screenshot 2021-03-08 161902](https://user-images.githubusercontent.com/6276915/110341260-71227f00-802a-11eb-9b9f-4dc04037c41c.png)


</details><br/>

<details><summary>Screenshot-1With AG

![Screenshot 2021-03-08 162032](https://user-images.githubusercontent.com/6276915/110341291-78498d00-802a-11eb-803f-eb54e9eb9a9b.png)


</details><br/>

**Comment by Morku** [comment]
@wout-codes 
Thanks, I have tried. But after the 11th check, I got the Ad again.

**Comment by Morku** [comment]
I also got sporadically Ads on the side. It's same rare as the other example. Takes about 5-10 tries.

<details><summary>Screenshot-1:</summary>

![kuchen](https://user-images.githubusercontent.com/6276915/110357631-c5822a80-803b-11eb-8c38-27ecb4a16247.png)


</details><br/>

<details><summary>"Why do I see the Ads?" </summary>

![kuchen2](https://user-images.githubusercontent.com/6276915/110357742-e77bad00-803b-11eb-94b5-fc90c99f74f5.png)


</details><br/>

- it's displayed because of my current search words.

**Comment by krystian3w** [comment]
I see two reasons:

- Google loads the ad with a new method every 10 "refresh" as part of an A/B test (then you have to come up with new an ExtendedCSS rule for you to test)
- The bug is in the performance of the AdGuard application on Windows in correlation with Firefox

**Comment by wout-codes** [comment]
@Morku Why don't you use Chrome and check?

**Comment by Morku** [comment]
@wout-codes 
Because I like Firefox for many reasons.

After several attempt (didn't count) I was also able to reproduce on Chrome:

<details><summary>Screenshot-1:</summary>

![Anmerkung 2021-03-09 091339](https://user-images.githubusercontent.com/6276915/110439717-159ed280-80b8-11eb-9f24-2145a6d08d20.png)


</details><br/>

**Comment by wout-codes** [comment]
Hmm Let me try again and check. @Alex-302 @zloyden Can you check too?

**Comment by Morku** [comment]
As usual, I also offer to check on my system with Anydesk in case it is IP geo related.

**Comment by Alex-302** [comment]
@Morku please copy HTML from dev tools and pase it for example here https://gist.github.com/
![image](https://user-images.githubusercontent.com/8361299/111329838-1a472600-8678-11eb-8dcf-6bbc7bb61097.png)

**Comment by Morku** [comment]
@Alex-302 
"Computer Leipzig": https://gist.github.com/Morku/225a8a524d6a9ab6500f1ea63ad3d330

I am unable to reproduce "kuchen berlin". As soon I see, I will post as well.

**Comment by Alex-302** [comment]
Hmm. Still don't see. 
@BlazDT Could you please check it?

**Comment by Morku** [comment]
"augenarzt halle": https://gist.github.com/Morku/613b391446e3a9faf26ac9c31ca04baa

<details><summary>Screenshot-1:</summary>

![augen](https://user-images.githubusercontent.com/6276915/112125111-5ec84980-8bc3-11eb-9c63-138c694b62dd.png)


</details><br/>

**Comment by krystian3w** [comment]
I don't see intresting attrib other than `.VqFMTc`, maybe this will work fast and with good performance for CPU:

`google.*#?#div[data-hveid] a[role="link"] span > span:only-child:contains(Anzeige):upward(div[data-hveid])`

Based on path and span structure:

`google.*#?#div[data-hveid] > div[jsname] > div[class] > a[role="link"] > div > span > span:only-child:upward(div[data-hveid])`

**Comment by BlazDT** [comment]
That entry is shown to me without ad tag. Normal CSS is not matching except first one (of course not showing Anzeige). So looks safe to me.

**Comment by Morku** [comment]
@BlazDT 
For "augenarzt halle", I agree to you. The "Institut für Augenheilkunde Halle" is also shown as first geo result, when not "Anzeige" is shown (oddly with another count of ratings).

But for "computer leipzig" it's not. When there is no "Anzeige" result, the first result is "K&M Computer". But when the "Anzeige" shows up, it's "Computer-Insel", but this is documented in my video (first post).
They look to me to priorized paid results.

**Comment by BlazDT** [comment]
Can you please try this rule:

`google.*#?##search div[data-hveid] div[jscontroller] div[jscontroller][data-hveid]:has(> div[jsname][jsaction] a[data-ved] div[role="heading"] + span[class] > span[class] > span[style="padding:0 5px"])`

It is not causing breakage for this issue for me:
https://github.com/AdguardTeam/AdguardFilters/issues/73503#issuecomment-774706270

**Comment by Morku** [comment]
@BlazDT 
I have tried about 50x "augenarzt halle" and didn't see any Anzeige Ad.

On the other hand, I disabled the rule and tried again about 30x. Still no Anzeige Ad.

So _maybe_ it's working? difficult to say at this random problem. I will try tomorrow again.

**Comment by BlazDT** [comment]
You can check in code if it is working.

**Comment by Morku** [comment]
@BlazDT 
sry for my knowledge. Can you give me a short hint, which Code? Where do I find?

**Comment by BlazDT** [comment]
Open browser's developing tools -> ctrl-f and paste  #search div[data-hveid] div[jscontroller] div[jscontroller][data-hveid]

If it has a style with something like display: none !important; then it is working.

**Comment by krystian3w** [comment]
Possible use `tabs.insertCSS()` https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/insertCSS like in AdGuard Addon 3.3.2 (I suppose someone disabled use this for ExtendedCSS)?

**Comment by Morku** [comment]
@BlazDT 
I can confirm, it's working!

<details><summary>Screenshot-1:</summary>

![augen](https://user-images.githubusercontent.com/6276915/112280481-58020b00-8c85-11eb-891e-133c8e6893e2.png)


</details><br/>

**Comment by wout-codes** [comment]
Nice.

**Comment by Morku** [comment]
Now I also was able to reproduce "kuchen berlin" (after 10 tries), but it's not on the right side, it's now on the top:

<details><summary>Screenshot-1:</summary>

![kuchen1](https://user-images.githubusercontent.com/6276915/112285537-b54c8b00-8c8a-11eb-8992-ff1909d6c062.png)


</details><br/>

https://gist.github.com/Morku/04772ae625b4b6bb52e96c0ced039343

**Comment by krystian3w** [comment]
also possible use `style="padding:0 5px"`

**Comment by BlazDT** [comment]
Added previously posted rule. Other one I cannot reproduce. Only sometimes an additional margin-top on first class "ULSxyf" to make room for ads likely.

`#main > div > style + div[id] > div[class] > div[class] > div[data-hveid]` and `###main > div > style + #atvcap` no match for me, even when additional margin-top is shown.

**Comment by Morku** [comment]
@BlazDT 
Anything I can test for the kuchen Ad? Shall I add krystian3w and check the Code for `#main > div > style + div[id] > div[class] > div[class] > div[data-hveid]` to see if hided again?

**Comment by krystian3w** [comment]
I'm working the saved file on localhosts, maybe that's where the weirdness of it having zero matches comes from real domain.

**Comment by krystian3w** [comment]
I also made a long path to reduce the amount of checking, because there are probably 50 div's with the attribute `[data-hveid]`.

**Comment by BlazDT** [comment]
@Morku 
Please try the rule posted by @krystian3w

**Comment by krystian3w** [comment]
@AdamWr  Google still use widget to compare smartphones? Or something like embeded specification if we try found information about "iPhone 12"?

**Comment by AdamWr** [comment]
>Google still use widget to compare smartphones? Or something like embeded specification if we try found information about "iPhone 12"?

I guess you mean this - https://github.com/AdguardTeam/AdguardFilters/issues/64437#issuecomment-697354307
If so, it's still used.

**Comment by Morku** [comment]
I can't reproduce the "kuchen berlin" Ad anymore. Not on the right side, not on the top. Since one week I tried daily about 30x.
`#main > div > style + div[id] > div[class] > div[class] > div[data-hveid]` never appeared again in the search result code. Who knows. 🤷‍♀️

At least, the geo result Ad is fixed. I was able to reproduce for "computer leipzig" again and it is gone thanks to the updated rule.

<details><summary>Screenshot-1:</summary>

![computer leipzig](https://user-images.githubusercontent.com/6276915/113288796-b36c7280-92ef-11eb-8d63-b46c1e76fe8a.png)

</details><br/>

**Comment by krystian3w** [comment]
Google started new consent pop-up and cookie wall:

<details> <summary> pop-up 🍯 </summary>

![Screenshot_2021-04-01 Google](https://user-images.githubusercontent.com/35370833/113291427-4eb31700-92f3-11eb-8b21-ff2401ab08b2.png)

</details>

`https://consent.google.de/m?continue=https://www.google.de/news&gl=DE&m=0&pc=srp&hl=de&src=1`

<details> <summary> cookie wall 🍯 </summary>

![Screenshot_2021-04-01 Bevor Sie zur Google Suche weitergehen](https://user-images.githubusercontent.com/35370833/113291714-b23d4480-92f3-11eb-8a6d-951af1192efa.png)


</details>

**Comment by Morku** [comment]
Yep, today I get it too, when I want to browse maps.google.de

<details><summary>Screenshot-1:</summary>

![maps](https://user-images.githubusercontent.com/6276915/113404911-3ac9ec80-93a9-11eb-8c47-0781ae993024.png)


</details><br/>

https://www.reddit.com/r/uBlockOrigin/comments/mi1azc/googles_before_you_continue_cookies_consent_is/
https://www.reddit.com/r/uBlockOrigin/comments/mieih9/do_you_believe_that_the_youtube_tracking_consent/
https://www.reddit.com/r/uBlockOrigin/comments/mi2mpv/how_do_i_block_the_new_google_consent/
https://www.reddit.com/r/uBlockOrigin/comments/mhqxeh/google_search_cookie_consent_popup_problem_again/
https://www.reddit.com/r/uBlockOrigin/comments/mhxl1w/before_you_continuer_to_youtube/
https://www.reddit.com/r/uBlockOrigin/comments/mh2su3/any_filter_that_stops_before_you_continue_to/

**Comment by krystian3w** [comment]
Few help these in AdGuard for Windows:

`||consent.youtube.com^$cookie,domain=consent.youtube.com`
`||consent.google.*^$cookie,domain=consent.google.*`

**Comment by Morku** [comment]
As I know, it will break comment function of YouTube.

**Comment by AdamWr** [comment]
I think that similar rule like for `youtube` should works:
```
consent.google.*#%#AG_onLoad(function(){if(window.location.href.includes("consent.google.")){var a=new MutationObserver(function(){var b=document.querySelector('form[action^="https://consent.google."][action$="/s"]');b&&(a.disconnect(),b.submit())});a.observe(document,{childList:!0,subtree:!0});setTimeout(function(){a.disconnect()},1E4)}});
```

**Comment by Morku** [comment]
@AdamWr 
Works for me :)

**Comment by AdamWr** [comment]
Ok, added.
Thank you for your reply.

**Comment by Morku** [comment]
@AdamWr 
For some reason, on Firefox it works fine, but not for Edge. But browsing youtube.com on Edge is fine. Strange


https://user-images.githubusercontent.com/6276915/113410974-ecbae600-93b4-11eb-9c08-607488f3305d.mp4

**Comment by AdamWr** [comment]
I modified a rule, so it should works fine in Edge too.
